sbi/simulators/mp_simutils.py

```diff
@@ -1,39 +1,46 @@
 import multiprocessing as mp
 import numpy as np
 from tqdm.auto import tqdm
-from typing import Callable
+from typing import Callable, Tuple
 import torch
 from sys import platform
+from torch import Tensor
 
 
 def simulate_mp(
     simulator: Callable,
-    theta: torch.Tensor,
+    theta: Tensor,
     num_workers: int = 4,
     worker_batch_size: int = 20,
     verbose: bool = False,
     show_progressbar: bool = True,
-) -> (torch.Tensor, torch.Tensor):
+) -> Tuple[Tensor, Tensor]:
     """
-    Wrapper function for the callable simulator to allow it to use multiprocessing.
+    Return parameters theta and simulated data x, simulated using multiprocessing on the
+        passed simulator.
 
     Args:
         simulator: simulator function that will be executed on each worker
         theta: all simulation parameters
         num_workers: number of parallel workers to start
-        worker_batch_size: how many thetas are processed on each worker. This number of
-            thetas will then be handled by simulate_in_batches()
+        worker_batch_size: Number of parameters that are processed per worker. A worker
+            will receive this many parameters to simulate per call. Needs to be larger
+            than sim_batch_size. A lower value creates overhead from starting workers
+            frequently. A higher value leads to the simulation progressbar being updated
```

Comment:
I like how candid the comment is about the tradeoff, but the frequent updating of the progress bar sounds somewhat like a frivolous reason ;D

Reply:
Well, the other reason is that if you have only very few simulations overall, you want a small value here. E.g. if you have 10 workers and 50 simulations in total, you want to set `worker_batch_size=5`. Does it make sense to add this to the docstring? Or is it too specific?